### PR TITLE
use mqtt v1 in docs

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Update documentation to use the tedge mqtt v1 API (e.g. `te/` topics instead of `tedge/`). The current scope of this PR is to only switch to using the new API and not do too much refactoring on the presentation of the docs.

Note, changes in the health endpoints are not included as the docs are being updated in a separate PR (https://github.com/thin-edge/thin-edge.io/pull/2375).

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2378

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/references/data-model.md
Comment (rina23q) [inline docs/src/references/data-model.md:184]: > must be `critical`, `major`, `minor` or `warning`

is no longer applicable. It accepts anything in cloud-agnostic level.

```suggestion
|`severity`          |a string part of the MQTT payload, that indicates the severity of the alarm; recommended to be `critical`, `major`, `minor` or `warning`|
```

File: docs/src/start/raise-alarm.md
Comment (rina23q) [inline docs/src/start/raise-alarm.md:68]: I would change these severities are the "recommendations".

```suggestion
The `<severity>` value in the MQTT payload is recommended to be one of the following values:
```

File: docs/src/understand/tedge-mapper.md
Comment (rina23q) [inline docs/src/understand/tedge-mapper.md:198]: It will be changed soon by [#2377](https://github.com/thin-edge/thin-edge.io/issues/2377)

File: docs/src/understand/thin-edge-json.md
Comment (rina23q) [inline docs/src/understand/thin-edge-json.md:217]:
```suggestion
| `severity`   | Severity of the alarm; recommended to be `critical`, `major`, `minor` or `warning`                                                                                           |
```

**Review by rina23q — CHANGES_REQUESTED** [review]
The severity explanation for alarm must be updated. It used to accept only 4 types, but we changed to accept any names in cloud-agnostic level.

File: docs/src/understand/tedge-mapper.md
Comment (reubenmiller) [reply docs/src/understand/tedge-mapper.md:198]: We'll update the `tedge/errors` topic in another PR so we can merge this as soon as possible.

**Comment by reubenmiller** [comment]
@rina23q Thanks for the suggestions, I've applied them as suggested. We'll hold off on the `tedge/errors` for now until it is actually merged as this change will be a lot easier in a separate PR.

File: docs/src/references/agent/software-management.md
Comment (didier-wenzek) [inline docs/src/references/agent/software-management.md:33]: Note, that currently these values are ignored.

File: docs/src/start/send-thin-edge-data.md
Comment (didier-wenzek) [inline docs/src/start/send-thin-edge-data.md:34]: The `environment` type is used by the example but not introduced at all.

**Review by didier-wenzek — APPROVED** [review]

File: docs/src/understand/tedge-mapper.md
Comment (didier-wenzek) [reply docs/src/understand/tedge-mapper.md:198]: The doc will be updated along the code: https://github.com/thin-edge/thin-edge.io/pull/2386

File: docs/src/references/agent/software-management.md
Comment (reubenmiller) [reply docs/src/references/agent/software-management.md:33]: Are you sure? Because I found this doc change when updating the tedge-container-demo to use the new MQTT api and ran into this issue. I changed the child device to use the `types` property instead of `supportedTypes` and the expected types were sent back to Cumulocity IoT as expected.

File: docs/src/start/send-thin-edge-data.md
Comment (reubenmiller) [reply docs/src/start/send-thin-edge-data.md:34]: Actually on careful reconsideration I added a note here cb2eebe33c9d2893230a822cfb68cce3499b7332

File: docs/src/references/agent/software-management.md
Comment (reubenmiller) [reply docs/src/references/agent/software-management.md:33]: I added a note about this future feature in bae425ccbf28f25d7749b05be339f3836a08e763

**Pull request merged**
